src/components/data/data-utils.js

```diff
@@ -1185,13 +1187,26 @@ export const ModalToolButton = ({ icon, text, disabled, ...props }) => {
   ])
 }
 
-export const HeaderOptions = ({ sort, field, onSort, extraActions, children }) => {
-  const columnMenu = h(MenuTrigger, {
+export const HeaderOptions = ({ sort, field, onSort, extraActions, renderSearch, searchByColumn, children }) => {
+  const popup = useRef()
+  const columnMenu = h(PopupTrigger, {
+    ref: popup,
     closeOnClick: true,
     side: 'bottom',
     content: h(Fragment, [
       h(MenuButton, { onClick: () => onSort({ field, direction: 'asc' }) }, ['Sort Ascending']),
       h(MenuButton, { onClick: () => onSort({ field, direction: 'desc' }) }, ['Sort Descending']),
+      renderSearch && h(ConfirmedSearchInput, {
+        'aria-label': 'Filter by column',
+        placeholder: 'Exact match search in column',
+        onChange: e => {
+          if (!!e) {
+            searchByColumn(e)
+            popup.current.close()
+          }
+        },
+        onClick: e => { e.stopPropagation() }
+      }),
```

Comment:
maybe give the search field just a tiny bit of margin? I know we don't want to lose too much horizontal space here.
```suggestion
      renderSearch && h(div, { style: { width: '98%' } }, [h(ConfirmedSearchInput, {
        'aria-label': 'Filter by column',
        placeholder: 'Exact match search in column',
        style: { marginLeft: '0.25rem' },
        onChange: e => {
          if (!!e) {
            searchByColumn(e)
            popup.current.close()
          }
        },
        onClick: e => { e.stopPropagation() }
      })]),
```

Reply:
Done